File: .archon/workflows/sdlc/lifecycle/scripts/select-qualified.py
"""Select an explicit initial or repaired qualification, without inferring run status."""
import json
import os

def value(name):
    return json.loads(os.environ['INPUTS_' + name.upper()])

initial = value('initial')
retry = value('retried')
candidate = value('candidate')
if initial['repair']:
    candidate = value('repaired')
    result = retry
else:
    result = initial
ready = bool(result and candidate and result['ready'] and candidate['delivered'])
print(json.dumps({'ready': ready, 'prs': candidate['prs'] if ready else [],
                  'evidence': result['evidence'] if result else initial['evidence']}))

File: .archon/workflows/sdlc/lifecycle/scripts/select-qualified.ts
function object(name: string): Record<string, unknown> | null {
  const value: unknown = JSON.parse(process.env[`INPUTS_${name}`] ?? 'null');
  if (value === null) return null;
  if (typeof value !== 'object' || Array.isArray(value)) throw new Error(`invalid ${name}`);
  return Object.fromEntries(Object.entries(value));
}

const initial = object('INITIAL');
if (initial === null || typeof initial.repair !== 'boolean') throw new Error('missing initial qualification');
const result = initial.repair ? object('RETRIED') : initial;
const candidate = object(initial.repair ? 'REPAIRED' : 'CANDIDATE');
const ready = result?.ready === true && candidate?.delivered === true;
console.log(JSON.stringify({ ready, prs: ready ? candidate.prs : [],
  evidence: result?.evidence ?? initial.evidence, references: ready ? result.references : [] }));
